### ISSUE TITLE
Parsing issue: StringLiteral being wrapped in a PrefixExpression when preceded by a metadata tag.

### ISSUE THREAD

**Issue by EricBishton** [HaxeFoundation/intellij-haxe#709]
Back when we originally fixed #81, we added an optional macroList to the value rule.  This allowed metadata in far more positions, but it had the side effect of forcing any value (pretty much and statement or expression) into a prefixExpression by virtue of the fact that prefixExpression is the first non-private rule to contain a value.

We need a solution that removes metadata from the regular syntax, much as comments are excluded.  They will still be there in the AST but should not affect the parsing rules or the PSI by their presence.  Most importantly, they should not have to be intermingled with the language rules.

**Comment by EricBishton** [comment]
For reference, this code:
```haxe
package ;
@:meta class TestIssue81 {
    @meta static function main() {
        @meta trace(@meta "Haxe is great!");
        var target = {pos:0}
        var targetpos = @meta @meta @:pos(target){ @:meta 10; }
    }
}
```
This part of the PSI shows the error:
```
          CALL_EXPRESSION
            REFERENCE_EXPRESSION
              IDENTIFIER
                HaxePsiToken:ID('trace')
            HaxePsiToken:(('(')
            EXPRESSION_LIST
              PREFIX_EXPRESSION              <<<<<<<< Shouldn't be here at all.
                PsiModifierList:@meta
                  MACRO_CLASS
                    CUSTOM_META
                      HaxePsiToken:MACRO_ID('@meta')
                STRING_LITERAL_EXPRESSION
                  HaxePsiToken:OPEN_QUOTE('"')
                  HaxePsiToken:REGULAR_STRING_PART('Haxe is great!')
                  HaxePsiToken:CLOSING_QUOTE('"')
            HaxePsiToken:)(')')
          HaxePsiToken:;(';')
```

**Comment by EricBishton** [comment]
Fixed with PR #995.  Was released with plugin version 1.3.
